### ISSUE TITLE
seed parameter where it is not requested

### ISSUE THREAD

**Issue by EugeniuStrelet** [uber/causalml#857]
**Describe the bug**
There is a parameter in DRLearner that it is not expected to be there. In drlearner.py file line 435, ther is requested seed parameter dor bootstrap method, when this paramenter is not defined in the method per se. It does not allow the BaseDRLearner to run properly. I commented it and it worked.

**To Reproduce**
learner_dr = BaseDRRegressor(XGBRegressor(), control_name='control')
ate_dr_b_no_p, ate_dr_lb_b_no_p, ate_dr_ub_b_no_p = learner_dr.estimate_ate(X=X, treatment=treatment, y=y, bootstrap_ci=True, n_bootstraps=100, bootstrap_size=5000, seed=42)

**Expected behavior**
To run and return ate_dr_b_no_p, ate_dr_lb_b_no_p, ate_dr_ub_b_no_p

**Screenshots**

<img width="611" height="340" alt="Image" src="https://github.com/user-attachments/assets/d75dd25f-d513-4ed0-bbfc-f93bcb992d86" />

**Environment (please complete the following information):**
 - OS: Windows 11
 - Python Version: 3.11
 - Versions of Major Dependencies (`pandas`, `scikit-learn`, `cython`):
    "causalml>=0.15.5",
    "fastparquet>=2024.11.0",
    "ipykernel>=7.1.0",
    "matplotlib>=3.10.7",
    "numpy>=2.3.4",
    "pandas>=2.3.3",
    "scikit-learn>=1.7.2",
    "seaborn>=0.13.2",
    "statsmodels>=0.14.5",
    "xgboost>=3.1.1",

**Additional context**
There is no additional context
